src/client/Microsoft.Identity.Client/Internal/ClientCredential/CertificateAndClaimsClientCredential.cs

```diff
@@ -42,7 +46,9 @@ public CertificateAndClaimsClientCredential(X509Certificate2 certificate, IDicti
                 tokenEndpoint,
                 _claimsToSign,
                 _appendDefaultClaims);
-            
+
+            _sendX5c = sendX5C;
```

Comment:
I don't think this will work.

Reply:
Oh right, I forgot that this is set at the application level so there can be interactions collisions between two requests.